src/include/network/services/dns.rb

```diff
@@ -294,6 +301,26 @@ def StoreSettings(settings)
       nil
     end
 
+    def current_hostname
+      config&.hostname&.static || ""
+    end
+
+    def update_hostname_hosts(hostname)
+      return false if hostname == current_hostname
+      return false if current_hostname.to_s.empty?
+
+      conn = config.connections.find { |c| c.hostnames.include?(current_hostname) }
+
+      update_hostname = conn && Popup.YesNo(
+        wrap_text(
+          format(_("The interface '%s' static IP address is mapped to the modified hostname.\n\n" \
+                "Would you like to adapt it now?"), conn.name)
```

Comment:
I would mention new hostname here to avoid any confusion.
So something like `"The interface '%s' static IP address is mapped to #{current_hostname}. Would you like to adapt it to #{hostname} now?` Also is possible that new hostname is empty?

Reply:
The IP address is mapped to the hostname before the change. The new hostname could be empty, but that is not a problem. The user decides if want to remove the hostname completely or not, I thought about adding an explicit 'delete' button, but currently works as the behavior in the address tab, so, looks consistent.